src/ReplicatedStorage/Client/InstanceComponents/Misc/CameraLook.lua

```diff
@@ -0,0 +1,164 @@
+end
+
+function CameraLook.new(instance: Instance)
+	local self = setmetatable({}, CameraLook)
+
+	self.Instance = instance
+	self._trove = Trove.new()
+
+	-- Wait for Humanoid to be added
+	local humanoid = instance:FindFirstChildWhichIsA("Humanoid")
+	while not humanoid do
+		humanoid = instance:FindFirstChildWhichIsA("Humanoid")
+		instance.ChildAdded:Wait()
+	end
+
+	-- Retrieve player root part
+	local rootPart = humanoid.RootPart
+
+	-- Speed for rotating the character itself
+	self.LookAngularSpeed = math.rad(400)
+
+	-- Disable automatic rotation
+	humanoid.AutoRotate = false
+
+	-- Speed for rotating the root
+	self.RootPitchSpeed = math.rad(80)
+	self.RootYawSpeed = math.rad(0)
+	-- Speed for rotating the waist
+	self.WaistPitchSpeed = math.rad(100)
+	self.WaistYawSpeed = math.rad(360)
+	-- Speed for rotating the neck
+	self.NeckPitchSpeed = math.rad(200)
+	self.NeckYawSpeed = math.rad(560)
+
+	-- Grab base C0 for root part (TODO: Clean this up)
+	local lowerTorso: BasePart = instance:WaitForChild("LowerTorso")
+	local root: Motor6D = lowerTorso and lowerTorso:WaitForChild("Root")
+	local rootC0 = root.C0
+
+	-- Reset joint angles
+	self._rootPitch = 0
+	self._rootYaw = 0
+	self._waistPitch = 0
+	self._waistYaw = 0
+	self._neckPitch = 0
+	self._neckYaw = 0
+
+	local player: Player? = Players:GetPlayerFromCharacter(instance)
+	local isLocalPlayer = player == Players.LocalPlayer
+	local mouse: Mouse? = if isLocalPlayer then player:GetMouse() else nil
+	self._trove:Connect(RunService.Stepped, function(_, deltaTime)
+		-- Get root CFrame & vectors
+		local cframe = rootPart:GetPivot()
+
+		local upVector = cframe.UpVector
+		local rightVector = cframe.RightVector
+
+		-- Locate mouse target part
+		local lookPart: BasePart = instance:FindFirstChild("MouseTarget")
+		local lookCFrame
+		if isLocalPlayer then
+			-- Use the mouse's CFrame
+			lookCFrame = mouse.Hit
+
+			-- Set target part CFrame
+			if lookPart then
+				lookPart.CFrame = lookCFrame
+			end
+		else
+			-- Use the look part's CFrame, and fall back to root's CFrame
+			if lookPart then
+				lookCFrame = lookPart.CFrame
+			else
+				lookCFrame = cframe
+			end
+		end
+
+		-- Determine the look position
+		local lookPosition = lookCFrame.Position
+
+		-- Root joint angles
+		local lowerTorso: BasePart? = instance:FindFirstChild("LowerTorso")
+		local root: Motor6D? = lowerTorso and lowerTorso:FindFirstChild("Root")
+		if root then
+			self._rootPitch = dampenAngle(self._rootPitch, calculateJointAngle(-upVector, lowerTorso.Position, lookPosition), self.RootPitchSpeed, deltaTime)
+			self._rootYaw = dampenAngle(self._rootYaw, calculateJointAngle(rightVector, lowerTorso.Position, lookPosition), self.RootYawSpeed, deltaTime)
+			local rootCFrame = CFrame.Angles(
+				math.clamp(self._rootPitch, math.rad(-55), math.rad(20)),
+				math.clamp(self._rootYaw, math.rad(-30), math.rad(30)),
+				0
+			)
+
+			root.Transform *= rootCFrame
```

Comment:
Perhaps this should be using to object space to be more familiar to contributors without a heavy math background.

Reply:
(I think)